src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TransactionParserImpl implements TransactionParser {
+    private static final String SEPARATOR = ",";
+    private static final int INDEX_CODE_OPERATION = 0;
+    private static final int INDEX_FRUIT_NAME = 1;
+    private static final int INDEX_FRUIT_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> parseData(List<String> data) {
+        if (data != null && data.isEmpty()) {
+            throw new RuntimeException("Not data found");
+        }
+        try {
+            return data.stream()
+                    .map(element -> element.split(SEPARATOR))
+                    .map(element -> new FruitTransaction(FruitTransaction.Operation
+                            .getOperationByCode(element[INDEX_CODE_OPERATION]),
+                            element[INDEX_FRUIT_NAME],
+                            Integer.parseInt(element[INDEX_FRUIT_QUANTITY])))

```

Comment:
add private method for this
```suggestion
                    .map(line-> parseLine(line))
```